x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/single_page_layout/index.tsx

```diff
@@ -447,6 +462,17 @@ export const CreatePackagePolicySinglePage: CreatePackagePolicyParams = ({
             onCancel={() => setFormState('VALID')}
           />
         )}
+        {isUnprivilegedModalOpen && !(formState === 'CONFIRM' && agentPolicy) ? (
```

Comment:
when I add auditd to an existing policy with an unprivileged agent is such that I get this modal first, I click `Add integration`, and I get the second `Save and deploy changes` confirmation modal. this seems like a UX anti-pattern.

is it possible to merge these two modals? since both modals are conditional on if there are already agents enrolled into the selected policy, we could have the unprivileged warning callout below the blue callout if there are some unprivileged agents:

<img width="804" alt="image" src="https://github.com/elastic/kibana/assets/1965714/23fec3e1-367e-448d-83b2-9d2c0afb30e6">

WDYT? and sorry for the review churn - I should have commented on this behavior in my initial review 😅

Reply:
Good point, showing two modals is not ideal indeed. I'll update.

Reply:
Merged the two modals:
<img width="807" alt="image" src="https://github.com/elastic/kibana/assets/90178898/420da729-a4f4-4861-9767-001699629397">